app/lib/differential_expression_service.rb

```diff
@@ -185,6 +153,93 @@ def self.run_differential_expression_job(cluster_file, study, user, annotation_n
     end
   end
 
+  # find all eligible annotations for DE for a given study
+  #
+  # * *params*
+  #   - +study+        (Study) => Associated study object
+  #   - +machine_type+ (String) => Override default VM machine type
+  #   - +dry_run+      (Boolean) => Indication of whether or not this is a pre-flight check
+  #   - +skip_existing+   (Boolean) => Skip annotations that already have DE results
+  #
+  # * *returns*
+  #   - (Array<Hash>) => Array of annotation objects available for DE
+  def self.find_eligible_annotations(study, machine_type: nil, dry_run: nil, skip_existing: false)
+    annotations = []
+    metadata = study.cell_metadata.where(annotation_type: 'group').select(&:can_visualize?)
+    annotations += metadata.map do |meta|
+      { annotation_name: meta.name, annotation_scope: 'study', machine_type: machine_type, dry_run: dry_run }
```

Comment:
```suggestion
      { annotation_name: meta.name, annotation_scope: 'study', machine_type:, dry_run: }
```

Reply:
Ah, thanks for noting.  It seems that rule was [recently added](https://github.com/broadinstitute/single_cell_portal_core/commit/1b98cc62f24e10a71cc20eeb7e61c632fff735f7) and I overlooked this implication.

Could you elaborate on the rationale for that rule?  Mixing reduces visual noise and is quite successful throughout our JavaScript.  So allowing it in Ruby seems like it'd make our code easier to read, and facilitate jumping across languages.  But maybe I'm missing a big downside.